Data to home page

portmil opened the pull request.

portmil committed:
Move token to userService

portmil committed:
Get recipes and categories from backend

portmil committed:
Make recipe's icon on home page show its category

portmil committed:
Fix user not logged in on signup

viivi-uhari commented on client/src/components/HomePage.js:
```diff
@@ -52,10 +54,13 @@ const HomePage = () => {
       <div key={recipe.id} className='recipe-card' onClick={() => navigate(`/${recipe.id}`)}>
         <div className='recipe-icon-container'>
-          <img className='recipe-icon' src={allIcon} alt='Icon for all'/>
+          <img className='recipe-icon' src={require(`../graphics/${icon}.svg`)} alt='Icon for the category of the recipe'/>
```
Could this alt text have the same format as above? `Icon for category '${category.name}'`

portmil replied:
Modified

viivi-uhari commented on client/src/components/HomePage.js:
```diff
@@ -15,11 +16,12 @@ const HomePage = () => {
+    );
+    categoryService.getAll().then(categories =>
+      setCategories(categories)
+    );
```
Could a `try catch` structure be implemented here?

portmil replied:
Modified

portmil committed:
Implement the changes proposed in PR comments

Modify the alt text of the category icon showed on recipe cards. Replace then syntax with try/catch and async/await when fetching recipes and categories.
